### ISSUE TITLE
[BUG] seq_aux_loss + pad_to_capacity doesn't work

### ISSUE THREAD

**Issue by SeunghyunSEO** [NVIDIA/Megatron-LM#1617]
The [routing_map](https://github.com/NVIDIA/Megatron-LM/blob/core_v0.12.1/megatron/core/transformer/moe/moe_utils.py#L585-L592) returned by `topk_softmax_with_capacity` is corrupted when `pad_to_capacity` is enabled.
routing_map should be filled with the number of expert_capacity true values, so there are dummy tokens, but this corrupted routing_map is directly used in the load-balancing calculation in [sequence_load_balancing_loss_func](https://github.com/NVIDIA/Megatron-LM/blob/core_v0.12.1/megatron/core/transformer/moe/moe_utils.py#L73).
The loss is further distorted because the code also feeds in the unfiltered gating scores collected [here](https://github.com/NVIDIA/Megatron-LM/blob/core_v0.12.1/megatron/core/transformer/moe/router.py#L265-L276).
As a result, the auxiliary load-balancing loss no longer reflects the true token distribution across experts.
especially, aux loss value become always 1.0.

when i pass non_padded_routing_map, and everything looks fine.

```python
    def seq_aux_loss_load_balancing(self, logits: torch.Tensor, bsz: int, seq_length: int):
        probs, routing_map, tokens_per_expert = topk_softmax_with_capacity(...)
        if self.training and torch.is_grad_enabled():
            # Apply sequence-auxiliary load balancing loss
            scores = self.compute_routing_scores_for_aux_loss(logits)
            """
            from https://arxiv.org/pdf/2412.19437
            L = aux_loss_coeff * sum_{i=1}^{num routed} f_i P_i
            f_i = {num routed}/{topk * T} sum_{t=1}^{T} (1 * s_{i,t})
            s'_{i,t} = softmax prob across num routed experts
            P_i = 1/T * sum_{t=1}^{T} s'_{i,t}
            """
            non_padded_routing_map = torch.logical_and(routing_map, (probs > 0.0)) # because padded tokens has 0.0 probs and softmax, sigmoid gaiting has always non-zero positive values
            aux_loss_func = partial(
                sequence_load_balancing_loss_func,
                probs=scores,
                routing_map=non_padded_routing_map, # fixed
                batch_size=bsz,
                seq_length=seq_length,
                topk=self.topk,
            )
            probs = self.apply_load_balancing_loss(
                activation=probs, load_balancing_loss_func=aux_loss_func
            )
```

here's my experimental results of 16B total MoE + GQA model.

```
 0:   moe_pad_expert_input_to_capacity ................ True
 0:   moe_aux_loss_coeff .............................. 0.01
 0:   moe_router_load_balancing_type .................. seq_aux_loss
 0:   moe_router_pre_softmax .......................... True
 0:   moe_router_score_function ....................... softmax
 0:   moe_router_topk ................................. 6
```

<img width="487" alt="Image" src="https://github.com/user-attachments/assets/069c423f-7625-4362-8210-10fee22a3dfe" />

<img width="858" alt="Image" src="https://github.com/user-attachments/assets/edeb4c64-f016-44f4-8fea-20e351ae6b69" />
<img width="979" alt="Image" src="https://github.com/user-attachments/assets/9e6d65cf-6999-4862-a0bb-242cfadc1347" />
<img width="980" alt="Image" src="https://github.com/user-attachments/assets/979dcde1-173f-4ea8-95fe-83c094af1460" />

- observations 
  - idk why loss between padded and non_padded experiments are slightly different 
  - padded version shows better efficiency (throughput)
  - original version shows worse maxvio (or num tokens in experts) value and its load balancing loss does not decrease at all
  - fixed ersion shows better maxvio (near 0) and load balancing loss changes during training.
  - non_padded and padded version's num tokens in experts and load balancing loss seems similar

**Comment by SeunghyunSEO** [comment]
oh, [this issue](https://github.com/NVIDIA/Megatron-LM/issues/1513) seems related as well.
we might need to modify the existing methods further, since token dropping still causes problems even when pad_to_capacity is set to False.

updated) though i didn't finish tests due to resource constraint, using routing_map before dropping (`topk_map` variable in `topk_softmax_with_capacity`) looks healthier.
intuitively, load balancing loss should be increased at first because it is not balanced at first but if we use the dropped `routing_map`, it could be lower at first because most of the tokens would be dropped.

<img width="879" alt="Image" src="https://github.com/user-attachments/assets/44424ad9-ff33-4aac-aa15-12731ecf2427" />

<img width="861" alt="Image" src="https://github.com/user-attachments/assets/75a8cec7-2b27-427a-b445-2ff7ccc99642" />

(in above figure, light green one is exp using routing_map before dropping and padding)

**Comment by Victarry** [comment]
Thanks for sharing the detailed experiments.
We plan to solve this issue by recalculating the plain topk routing map in `compute_routing_scores_for_aux_loss` without considering token dropping or group limited routing. Do you think this is a reasonable solution?

**Comment by SeunghyunSEO** [comment]
> Thanks for sharing the detailed experiments. We plan to solve this issue by recalculating the plain topk routing map in `compute_routing_scores_for_aux_loss` without considering token dropping or group limited routing. Do you think this is a reasonable solution?

Yes, definitely! recalculating could be a good solution if there's no overhead. (or we can reuse `topk_map` from  `topk_softmax_with_capacity` i guess)

By the way — speaking of group-limited routing — I'm not sure how we're handling device or node limited LBL now..?
i guess only two out of the three type of aux losses (vanilla expert load balancing and sequence-wise load balancing losses) are currently implemented, right?

<img width="563" alt="Image" src="https://github.com/user-attachments/assets/04a26a65-ab52-41cf-8bc1-5c5242bc9585" />
<img width="470" alt="Image" src="https://github.com/user-attachments/assets/7b08126d-4602-4160-b627-c86303dc5b9a" />
<img width="559" alt="Image" src="https://github.com/user-attachments/assets/3fc40204-4041-41b5-a73f-1ed4800f72f8" />

**Comment by Victarry** [comment]
Right. Since DS-V3 didn't use the device-level balance loss, we are not going to add its support for now.
